wordpress-mobile/WordPress-iOS
# Issues/4033 reader detail refresh

Closes #4033 

This PR introduces an updated detail view for the reader.  The approach includes a switch to Swift for the language and a Storyboard for the UI.  This deprecates the old reader view hierarchy and a follow up PR will delete the old detail vc and its related code. 

New features introduced include:
- a revised UI (cc @shauwnandrews)
- showing/hiding parts of the UI when scrolling on the iPad
- an animated effect when toggling likes 
- on the iPad, the detail follows Apple's recommended readable margins and has an all white background

To Test:
Scenario 1
- Test viewing the detail from the post list. 

Scenario 2
- Test viewing the detail from a push notification

Scenario 3
- Test viewing the detail from a notification the Notifications tab

Scenario 4
- Test viewing the detail from the comments view controller. 
- Confirm that a detail opened this way does not show the comments button. 

Scenario 5 - Features
- Test liking a post. Confirm its animation plays
- Test viewing comments
- Test tapping on a tag to preview the tag topic
- Test tapping on a link to view the link in an in-app browser
- Test tapping on a featured image to view its image detail
- Test tapping on an image in the content to view its image detail
- Test playing a video in the content 
- Test sharing from the nav bar
- Test opening the menu
- Test blocking a site (you should navigate away from the post)
- Test previewing a site by tapping on the header
- Test that the preview site feature is disabled when viewing a post detail for a previewed site topic
- Test scrolling down hides the navbar and the footer view, and scrolling up reveals them again. 
- Test that content is resized correctly when the device changes orientation. 

Example:
![simulator screen shot dec 15 2015 5 32 40 pm](https://cloud.githubusercontent.com/assets/1435271/11827575/c329ec44-a352-11e5-8306-e3730c7f27d2.png)

This feature should be merged with the version 5.10 / 6.0 release (whichever we decide to make the milestone).

Needs review: @jleandroperez
